extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/devconsole/DevConsoleProcessor.java

```diff
@@ -32,61 +31,76 @@ public DevConsoleRuntimeTemplateInfoBuildItem collectScores() {
     public DevConsoleTemplateInfoBuildItem collectStaticResourcesInfo(
             ApplicationArchivesBuildItem applicationArchivesBuildItem) throws Exception {
 
-        Set<String> knownPaths = new TreeSet<>();
+        StaticResourceInfo staticResourceInfo = new StaticResourceInfo();
+
         for (ApplicationArchive i : applicationArchivesBuildItem.getAllApplicationArchives()) {
             Path resource = i.getChildPath(StaticResourcesRecorder.META_INF_RESOURCES);
             if (resource != null && Files.exists(resource)) {
-                collectKnownPaths(resource, knownPaths);
+                collectKnownPaths(resource, staticResourceInfo);
             }
         }
 
         ClassPathUtils.consumeAsPaths(StaticResourcesRecorder.META_INF_RESOURCES, resource -> {
-            collectKnownPaths(resource, knownPaths);
+            collectKnownPaths(resource, staticResourceInfo);
         });
-
-        return new DevConsoleTemplateInfoBuildItem("staticResourcesInfo", knownPaths);
+        return new DevConsoleTemplateInfoBuildItem("staticResourceInfo", staticResourceInfo);
     }
 
-    private void collectKnownPaths(Path resource, Set<String> staticResources) {
+    private void collectKnownPaths(Path resource, StaticResourceInfo staticResourceInfo) {
+
         try {
             Files.walkFileTree(resource, new SimpleFileVisitor<Path>() {
+                final Stack<StaticResourceInfo.StaticFile> currentFolder = new Stack<>();
 
                 @Override
-                public FileVisitResult visitFile(Path p, BasicFileAttributes attributes) throws IOException {
-                    String simpleName = p.getFileName().toString();
-                    String file = resource.relativize(p).toString();
-                    if (simpleName.equals("index.html") || simpleName.equals("index.htm")) {
-                        Path parent = resource.relativize(p).getParent();
-                        if (parent == null) {
-                            staticResources.add("/");
-                        } else {
-                            String parentString = parent.toString();
-                            if (!parentString.startsWith("/")) {
-                                parentString = "/" + parentString;
-                            }
-                            staticResources.add(parentString + "/");
-                        }
+                public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+
+                    Path relativeFile = resource.relativize(dir);
+                    String relativeFileString = "/" + relativeFile.toString();
+                    Path dirName = dir.getFileName();
+                    StaticResourceInfo.StaticFile getStaticFile = staticResourceInfo.resourceMap.get(relativeFileString);
+
+                    StaticResourceInfo.StaticFile staticFile = new StaticResourceInfo.StaticFile(dirName.toString(), true);
```

Comment:
You should only create a new `StaticFile` if `getStaticFile` is `null`.

Reply:
Yeah agree, I did that before but was getting some error + warning (`staticFile` not initialised ) bacause later after `if` i will be using that. But let me try again :)

Reply:
The general pattern is:

```java
Foo f = map.get(x);
if(f == null){
 f = new Foo();
 map.put(x, f);
}
// use f
```

Reply:
ah wait, on following this pattern I see I am getting same child 2 times.
like in root `/` , inside `children` instead of 4 I am getting 8 children static Files.

Reply:
Yes, if you did not need to create the folder, you don't need to add it to its parent :)

Reply:
ok that means only when I am creating new folder then only i need to add it to it's parent 
 ```

                    if (getStaticFile == null) {
                        getStaticFile = new StaticResourceInfo.StaticFile(dirName.toString(), true);
                        staticResourceInfo.resourceMap.put(relativeFileString, getStaticFile);

                        // adding to parent
                        if (!currentFolder.isEmpty()) {
                            currentFolder.peek().children.add(getStaticFile);
                        }
                    }

Reply:
Done